File: src/main/java/coza/trojanc/receipt/template/fields/Line.java
package coza.trojanc.receipt.template.fields;

import java.util.ArrayList;
import java.util.List;

/**
 * Class that represents a line within a template
 */
public class Line implements TemplateLine {

	private final List<TemplateTextItem> lineItems;

	public Line(){
		lineItems = new ArrayList<>();
	}

	public List<TemplateTextItem> getLineItems() {
		return lineItems;
	}

	public void addLineItem(TemplateTextItem item){
		lineItems.add(item);
	}

	@Override
	public String toString() {
		StringBuilder sb = new StringBuilder()
		.append("Line[");
		lineItems.forEach(templateTextItem -> {
			sb.append(templateTextItem.toString());
		});
		sb.append("]");
		return sb.toString();
	}
}

File: src/main/java/coza/trojanc/receipt/template/fields/Line.kt
package coza.trojanc.receipt.template.fields

import java.util.ArrayList

/**
 * Class that represents a line within a template.
 * @author Charl Thiem
 */
class Line : TemplateLine {

    private val lineItems: MutableList<TemplateTextItem>

    init {
        lineItems = ArrayList()
    }

    fun getLineItems(): List<TemplateTextItem> {
        return lineItems
    }

    fun addLineItem(item: TemplateTextItem) {
        lineItems.add(item)
    }

    override fun toString(): String {
        val sb = StringBuilder()
                .append("Line[")
        lineItems.forEach { templateTextItem -> sb.append(templateTextItem.toString()) }
        sb.append("]")
        return sb.toString()
    }
}
